glean-core/tests/counter.rs

```diff
@@ -0,0 +1,152 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+mod common;
+use crate::common::*;
+
+use glean_core::metrics::*;
+use glean_core::storage::StorageManager;
+use glean_core::{CommonMetricData, Glean, Lifetime};
+
+// SKIPPED from glean-ac: counter deserializer should correctly parse integers
+// This test doesn't really apply to rkv
+
+#[test]
+fn counter_serializer_should_correctly_serialize_counters() {
+    let (_t, tmpname) = tempdir();
+    {
+        let glean = Glean::new(&tmpname, GLOBAL_APPLICATION_ID).unwrap();
+
+        let metric = CounterMetric::new(CommonMetricData {
+            name: "counter_metric".into(),
+            category: "telemetry".into(),
+            send_in_pings: vec!["store1".into()],
+            disabled: false,
+            lifetime: Lifetime::User,
+            ..Default::default()
+        });
+
+        metric.add(&glean, 1);
+
+        let snapshot = StorageManager
+            .snapshot_as_json(glean.storage(), "store1".into(), true)
+            .unwrap();
+        assert_eq!(
+            "{\"counter\":{\"telemetry.counter_metric\":1}}",
+            ::serde_json::to_string(&snapshot).unwrap(),
+        );
+    }
+
+    // Make a new glean instance here, which should force reloading of the data from disk
+    // so we can ensure it persisted, because it has User lifetime
+    {
+        let glean = Glean::new(&tmpname, GLOBAL_APPLICATION_ID).unwrap();
```

Comment:
Ah, we need a new glean object (re `new_glean` from above).
Maybe we expose both things and in this test create the Glean objects ourselves, but otherwise use `new_glean` taking care of that?